compiler/noirc_frontend/src/hir/comptime/interpreter/cast.rs

```diff
@@ -267,7 +252,8 @@ mod tests {
             // Widen negative: zero extend
             (Value::I8(-1), unsigned(Sixteen), Value::U16(65535)),
```

Comment:
I'm just re-reading your explanation of extension [here](https://github.com/noir-lang/noir/pull/8469#issuecomment-2880842026) and wondering: when we go from `-1: i8` which is represented as `255: u8` to `u16` and end up with `65535`, which is `11111111 11111111`, aren't we _sign_ extending?

Reply:
Ah yes, the comment is wrong. We sign extend `signed -> unsigned` integer conversions, we only zero extend signed integers for `signed -> field`